src/lib/var.c

```diff
@@ -64,12 +64,17 @@ HashTable *xdebug_objdebug_pp(zval **zval_pp, int *is_tmp)
 		XG_BASE(in_debug_info) = 1;
 		orig_exception = EG(exception);
 		EG(exception) = NULL;
-
+		
+#if PHP_VERSION_ID >= 70400
+		*is_tmp = 0;
```

Comment:
You don't have to set `is_tmp` here, as the value is not (going to be) used. Instead of just using the `xdebug_var_maybe_destroy_ht`  in/around line 433 (under the `case XF_ST_OBJ_PROPERTY`), it should instead use the `zend_release_properties` function as is shown in/around line 497 (near `get_splobjectstorage_storage`), where there is an `#ifdef` for PHP 7.4 and higher to use `zend_release_properties` which is the proper way of "freeing" the memory "allocated" by `zend_get_properties_for`